src/Uno.UI/UI/Xaml/Media/ImageBrush.cs

```diff
@@ -65,14 +69,32 @@ public Stretch Stretch
 		#region ImageSource DP
 		public static DependencyProperty ImageSourceProperty { get; } =
 			DependencyProperty.Register("ImageSource", typeof(ImageSource), typeof(ImageBrush), new FrameworkPropertyMetadata(defaultValue: null, propertyChangedCallback: (s, e) =>
-			((ImageBrush)s).OnSourceChangedPartial((ImageSource)e.NewValue, (ImageSource)e.OldValue)));
+			((ImageBrush)s).OnSourceChanged((ImageSource)e.NewValue, (ImageSource)e.OldValue)));
 
 		public ImageSource ImageSource
 		{
 			get => (ImageSource)this.GetValue(ImageSourceProperty);
 			set => this.SetValue(ImageSourceProperty, value);
 		}
 
+		private void OnSourceChanged(ImageSource newValue, ImageSource oldValue)
+		{
+			if (newValue is null)
+			{
+				_sourceDisposable.Disposable = null;
+			}
+			else if (newValue is BitmapImage bitmapImage)
+			{
+				_sourceDisposable.Disposable = bitmapImage.RegisterDisposablePropertyChangedCallback(
+					BitmapImage.UriSourceProperty, (_, _) =>
+					{
+						OnSourceChangedPartial(newValue, null);
+					});
+			}
```

Comment:
if we transition from BitmapImage to something not listed here(eg: SvgImageSource), the old one isnt disposed
you should always dispose, and then set a new one if possible:
```suggestion
			_sourceDisposable.Disposable = null;
			if (newValue is BitmapImage bitmapImage)
			{
				_sourceDisposable.Disposable = bitmapImage.RegisterDisposablePropertyChangedCallback(
					BitmapImage.UriSourceProperty, (_, _) =>
					{
						OnSourceChangedPartial(newValue, null);
					});
			}
```